servicetalk-http-api/src/main/java/io/servicetalk/http/api/HttpClientBuilder.java

```diff
@@ -36,6 +37,14 @@
      */
     HttpClientBuilder<U, R, SDE> ioExecutor(IoExecutor ioExecutor);
 
+    /**
+     * Sets the {@link Executor} for all connections created from this builder.
+     *
+     * @param executor {@link Executor} to use.
+     * @return {@code this}.
+     */
+    HttpClientBuilder<U, R, SDE> executor(Executor executor);
```

Comment:
new methods on interfaces is API breaking, so we shouldn't merge into 0.41. if that is the case does it need to be part of this PR or can it come in #1695 for 0.42?

Reply:
The executor additions are already currently part of #1695. This change would simply add the methods sooner so that users can move from the deprecated methods now. The interfaces aren't ones that users implement or they extend only base classes that include the necessary implementation so I don't think there is need to worry about binary incompatibility.

Reply:
For 0.41, we should be cautious. This interface is pkg-private. For public abstract builders we can add a non-abstract method with noop default implementation. @chemicL is going to convert abstract classes into interfaces in 0.42. So, it will be cleaned and refactored in 0.42 anyway.

Reply:
I have added no-op implementations to the abstract builders so that any class which extends them will not fail with `AbstractMethodError`. If invoked the methods will generate a log warning.

Reply:
I have replaced the no-op implementations with versions that throw `UnsupportedOperationException` if used.